allennlp/predictors/sentence_tagger.py

```diff
@@ -32,3 +34,59 @@ def _json_to_instance(self, json_dict: JsonDict) -> Instance:
         sentence = json_dict["sentence"]
         tokens = self._tokenizer.split_words(sentence)
         return self._dataset_reader.text_to_instance(tokens)
+
+    @overrides
+    def predictions_to_labeled_instances(self,
+                                         instance: Instance,
+                                         outputs: Dict[str, np.ndarray]) -> List[Instance]:
+        """
+        Imagine an NER model predicts three named entities (each one with potentially
+        multiple tokens). For each individual entity, we create a new Instance that has
+        the label set to only that entity and the rest of the tokens are labeled as outside.
+        We then return a list of those Instances.
+
+        For example:
+        Mary  went to Seattle to visit Microsoft Research
+        U-Per  O    O   U-Loc  O   O     B-Org     L-Org
+
+        We create three instances.
+        Mary  went to Seattle to visit Microsoft Research
+        U-Per  O    O    O     O   O       O         O
+
+        Mary  went to Seattle to visit Microsoft Research
+        O      O    O   U-LOC  O   O       O         O
+
+        Mary  went to Seattle to visit Microsoft Research
+        O      O    O    O     O   O     B-Org     L-Org
+        """
+        predicted_tags = outputs['tags']
+        predicted_spans = []
+
+        i = 0
+        while i < len(predicted_tags):
+            tag = predicted_tags[i]
+            # if its a U, add it to the list
+            if tag[0] == 'U':
+                current_tags = [t if idx == i else 'O' for idx, t in enumerate(predicted_tags)]
+                predicted_spans.append(current_tags)
+            # if its a B, keep going until you hit an L.
+            elif tag[0] == 'B':
+                begin_idx = i
+                while tag[0] != 'L':
+                    i += 1
+                    tag = predicted_tags[i]
+                end_idx = i
+                current_tags = [t if idx >= begin_idx and idx <= end_idx else 'O' \
+                    for idx, t in enumerate(predicted_tags)]
+                predicted_spans.append(current_tags)
+            i += 1
+
+        # Creates a new instance for each contiguous tag
+        instances = []
+        for span in predicted_spans:
+            new_instance = deepcopy(instance)
+            new_instance.add_field('tags', \
+                ListField([LabelField(tag) for tag in span]), self._model.vocab)
```

Comment:
Use a `SequenceLabelField` here, instead of a `ListField[LabelField]`.  You'll have to grab the `TextField` and pass it in also, but it'll actually match what the dataset readers do, and make types better.